metadata-ingestion/src/datahub/configuration/time_window_config.py

```diff
@@ -42,35 +44,59 @@ class BaseTimeWindowConfig(ConfigModel):
     # if those fields are not set by the user.
     end_time: datetime = Field(
         default_factory=lambda: datetime.now(tz=timezone.utc),
-        description="Latest date of usage to consider. Default: Current time in UTC",
+        description="Latest date of lineage/usage to consider. Default: Current time in UTC",
     )
-    start_time: datetime = Field(default=None, description="Earliest date of usage to consider. Default: Last full day in UTC (or hour, depending on `bucket_duration`)")  # type: ignore
+    start_time: Union[datetime, str] = Field(default=None, description="Earliest date of lineage/usage to consider. Default: Last full day in UTC (or hour, depending on `bucket_duration`). You can also specify relative time with respect to end_time such as '-7 days' Or '-7d'.")  # type: ignore
 
     @pydantic.validator("start_time", pre=True, always=True)
     def default_start_time(
-        cls, v: Any, *, values: Dict[str, Any], **kwargs: Any
-    ) -> datetime:
-        return v or get_time_bucket(
-            values["end_time"] - get_bucket_duration_delta(values["bucket_duration"]),
-            values["bucket_duration"],
-        )
+        cls, v: Any, values: Dict[str, Any], **kwargs: Any
+    ) -> Union[datetime, str]:
+        if v is None:
+            return get_time_bucket(
+                values["end_time"]
+                - get_bucket_duration_delta(values["bucket_duration"]),
+                values["bucket_duration"],
+            )
+        elif isinstance(v, str):
+            # This is where start_time str is resolved to datetime
+            try:
+                delta = parse_relative_timespan(v)
+                assert delta < timedelta(
+                    0
+                ), "Relative start time should start with minus sign (-) e.g. '-2 days'."
+                assert abs(delta) > get_bucket_duration_delta(
+                    values["bucket_duration"]
+                ), "Relative start time should be in terms of configured bucket duration. e.g '-2 days' or '-2 hours'."
+                return values["end_time"] + delta
+            except humanfriendly.InvalidTimespan:
+                return parse_absolute_time(v)
+
+        return v
 
     @pydantic.validator("start_time", "end_time")
-    def ensure_timestamps_in_utc(cls, v: datetime) -> datetime:
-        if v.tzinfo != timezone.utc:
-            raise ValueError(
-                'timezone is not UTC; try adding a "Z" to the value e.g. "2021-07-20T00:00:00Z"'
-            )
+    def ensure_timestamps_in_utc(cls, v: Union[datetime, str]) -> Union[datetime, str]:
+        assert isinstance(v, datetime)
+        assert (
+            v.tzinfo == timezone.utc
+        ), 'timezone is not UTC; try adding a "Z" to the value e.g. "2021-07-20T00:00:00Z"'
+
         return v
 
+    @property
+    def parsed_start_time(self) -> datetime:
```

Comment:
Not a huge fan of this pattern

Given that the validators are `pre=True`, we should be able to keep the python type of start_time as `datetime`?

Reply:
The other option is to do something inline, but it feels less clean

```
if TYPE_CHECKING:
	start_time: datetime
else:
	start_time: Union[datetime, str] = ...
```

Reply:
using datetime with validator(pre=True) looks good to me. I've made changes accordingly.